StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+DeferredAPI.swift

```diff
@@ -97,13 +97,17 @@ extension PaymentSheet {
       try await withCheckedThrowingContinuation { continuation in
 
           if let confirmHandlerForServerSideConfirmation = intentConfig.confirmHandlerForServerSideConfirmation {
-              confirmHandlerForServerSideConfirmation(paymentMethodID, shouldSavePaymentMethod, { result in
-                  continuation.resume(with: result)
-              })
+              DispatchQueue.main.async {
+                  confirmHandlerForServerSideConfirmation(paymentMethodID, shouldSavePaymentMethod, { result in
+                      continuation.resume(with: result)
+                  })
+              }
           } else if let confirmHandler = intentConfig.confirmHandler {
-              confirmHandler(paymentMethodID, { result in
-                  continuation.resume(with: result)
-              })
+              DispatchQueue.main.async {
```

Comment:
In the CSC case we didn't see these errors since we were already running on the main thread and also not interacting with any UI in the "merchant app".

Reply:
Huh. This is a good change, we should have been calling merchant callbacks on the main thread, but why didn't it fail locally?

Reply:
I wonder if it has to do with a setting we are enabling for tests on CI that cause these warnings to be treated like errors.

Reply:
Link to that setting?

Reply:
GCC_TREAT_WARNINGS_AS_ERRORS is set to YES on the config file, it should be the same locally and in CI

Reply:
I don't know the setting (or if there is one). But I've seen Xcode crash or not crash on these types of memory warnings before. It's not clear to me why it crashes vs when it doesn't.